app/assets/src/components/views/amr_heatmap/AMRHeatmapVis.jsx

```diff
@@ -104,10 +99,22 @@ export default class AMRHeatmapVis extends React.Component {
     return heatmapValues;
   }
 
-  renderHeatmap() {
+  prepareHeatmapData() {
```

Comment:
I'd recommend calling this function `updateHeatmap` and renaming `renderHeatmap` to `initializeHeatmap`, to better describe what the functions are doing.

`render` typically suggests React rendering to me.